### PR TITLE
Upgrade test client

### PR DESCRIPTION
**Pull request by yunstanford**
upgrade aiohttp version for `sanic.test_client`

### CODE REVIEW

File: sanic/testing.py [hunk 1]
```diff
@@ -54,6 +56,15 @@ def _collect_request(request):
                     results[0] = request
             self.app.request_middleware.appendleft(_collect_request)
 
+        @self.app.exception(MethodNotSupported)
+        async def error_handler(request, exception):
```
Comment (yunstanford) [inline sanic/testing.py]: workaround fix for an aiohttp httpparser issue.

File: tests/test_keep_alive_timeout.py [hunk 2]
```diff
@@ -15,17 +15,30 @@ def __init__(self, *args, **kwargs):
...
-        if self.old_proto is not None:
-            if self.old_proto != new_conn._protocol:
-                raise RuntimeError(
-                    "We got a new connection, wanted the same one!")
-        print(new_conn.__dict__)
-        self.old_proto = new_conn._protocol
-        return new_conn
+    if aiohttp.__version__ >= '3.0':
+        @asyncio.coroutine
+        def connect(self, req, traces=None):
```
Comment (yunstanford) [inline tests/test_keep_alive_timeout.py]: kw arg `traces` only exists in `>=3.0`

File: tests/test_request_timeout.py [hunk 3]
```diff
@@ -80,18 +80,32 @@ def __init__(self, *args, **kwargs):
...
-        conn = yield from super(DelayableTCPConnector, self).connect(req)
-        if self._post_connect_delay and self._post_connect_delay > 0:
-            _ = yield from asyncio.sleep(self._post_connect_delay,
-                                         loop=self._loop)
-        req.send = d_req.send
-        t = req.loop.time()
-        print("Connected at {}".format(t), flush=True)
-        return conn
+    if aiohttp.__version__ >= '3.0':
+        @asyncio.coroutine
```
Comment (r0fls) [inline tests/test_request_timeout.py]: why are you using this syntax vs `async def`?

Comment (yunstanford) [reply tests/test_request_timeout.py]: @r0fls it was there before, i just moved it into the `if`, but we can change it to `async def`

Comment (yunstanford) [reply tests/test_request_timeout.py]: @r0fls done, interesting travici is failing, but it's passing here https://travis-ci.org/yunstanford/sanic

Comment (r0fls) [reply tests/test_request_timeout.py]: Ya weird, I thought this was passing before, and that change shouldn't impact it... let me try rerunning it

Comment (yunstanford) [reply tests/test_request_timeout.py]: yeah, it's passing both locally and here https://travis-ci.org/yunstanford/sanic..

**Pull request merged**